test/jdk/java/awt/datatransfer/Independence/IndependenceAWTTest.java

```diff
@@ -65,46 +91,44 @@ public void focusGained(FocusEvent fe) {
         tf2 = new TextField();
         tf3 = new TextField();
 
-        panel = new Panel();
+        Panel panel = new Panel();
         panel.setLayout(new BorderLayout());
 
         panel.add(tf2, BorderLayout.NORTH);
         panel.add(tf3, BorderLayout.SOUTH);
 
         frame.add(tf1, BorderLayout.NORTH);
         frame.add(panel, BorderLayout.CENTER);
+        frame.setLocationRelativeTo(null);
 
         frame.setVisible(true);
         tf1.requestFocus();
     }
 
     // Get System Selection i.e. Primary Clipboard
-    private void getPrimaryClipboard() {
-        Properties ps = System.getProperties();
-        String operSys = ps.getProperty("os.name");
+    private static void getPrimaryClipboard() {
         try {
-            pClip = Toolkit.getDefaultToolkit().getSystemSelection();
-            if (pClip == null)
-                if ((operSys.substring(0,3)).equalsIgnoreCase("Win") || operSys.toLowerCase().contains("os x"))
-                    System.out.println(operSys + "Operating system does not support system selection ");
-                else
-                    throw new RuntimeException("Method getSystemSelection() is returning null on X11 platform");
-        } catch(HeadlessException e) {
+            primaryClip = Toolkit.getDefaultToolkit().getSystemSelection();
+            if (primaryClip == null) {
+                throw new RuntimeException("Method getSystemSelection() is returning null"
+                                           + " on Linux platform");
```

Comment:
In previous version it was mentioning X11 platform. Do we know if the system selection works the same way on Wayland as it was on X11?

Reply:
Good point. The tests seem to work the same way in both X11 and Wayland. I'm looking into getSystemSelection() and if there is any display server specific code related to clipboards.

Currently tests look stable when I run on Wayland locally. It might be good to test it on CI system as well since these tests had stability issues.

Reply:
@azuev-java getSystemSelection() works the same for X11 and Wayland (goes through XToolkit's getSystemSelection() for both). I think the original test was added before wayland was introduced (going by the test's copyright year 1999, no associated `@bug` tag though) hence x11 was mentioned explicitly.